packages/astro/src/vite-plugin-scanner/index.ts

```diff
@@ -43,12 +44,15 @@ export default function astroScannerPlugin({
 			if (typeof pageOptions.prerender === 'undefined') {
 				pageOptions.prerender = defaultPrerender;
 			}
+			const fileExtension = extname(filename);
 
 			// `getStaticPaths` warning is just a string check, should be good enough for most cases
 			if (
 				!pageOptions.prerender &&
 				isServerLikeOutput(settings.config) &&
-				code.includes('getStaticPaths')
+				code.includes('getStaticPaths') &&
+				// this should only be valid for `.astro` files
+				fileExtension === '.astro'
```

Comment:
nit: I think it'll be slightly more performant if we only call `extname` inline here, incase the other conditions short-circuit.